File: sdks/php/src/Events/Requests/BatchEventsDto.php
<?php

namespace NovaBilling\Events\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Types\CreateEventDto;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class BatchEventsDto extends JsonSerializableType
{
    /**
     * @var array<CreateEventDto> $events Array of events to ingest (max 100)
     */
    #[JsonProperty('events'), ArrayType([CreateEventDto::class])]
    public array $events;

    /**
     * @param array{
     *   events: array<CreateEventDto>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->events = $values['events'];
    }
}

File: sdks/csharp/src/NovaBilling/Events/Requests/BatchEventsDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record BatchEventsDto
{
    /// <summary>
    /// Array of events to ingest (max 100)
    /// </summary>
    [JsonPropertyName("events")]
    public IEnumerable<CreateEventDto> Events { get; set; } = new List<CreateEventDto>();

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
